### PR TITLE
Stats plugin

### PR DESCRIPTION
**Pull request by mzieg**
- updated Stats to functional plugin
- improved default plugin plot x-axis

### CODE REVIEW

**Comment by mzieg** [comment]
Resolves #189.

File: pluginExamples/EnlightenPlugin.py [hunk 1]
```diff
@@ -88,9 +88,19 @@ def plot(self, y, x=None, title=None, color=None):
         @param title plot title, shown in legend
         @param color color of plot line
         """
+        if y is None:
```
Comment (samiebee43) [inline pluginExamples/EnlightenPlugin.py:91]: Please move this check to the caller.
I'm planning to support `plot(title="something", color="orange")` as a way to add arbitrary entries to the legend.

It's useful in LocalBaseline where I want to have n lines, but only one legend entry per color:
```
self.plot(title='X', color='red')
for i in range(n):
    self.plot(x=[...], y=[..])
```

**Review by samiebee43 — CHANGES_REQUESTED** [review]
See comment on line 91 of EnlightenPlugin.py

File: pluginExamples/EnlightenPlugin.py (hunk 1 above)
Comment (mzieg) [reply pluginExamples/EnlightenPlugin.py]: fixed

**Review by samiebee43 — APPROVED** [review]

**Pull request merged**